ruby/red-arrow/doc/text/examples.md

```diff
@@ -0,0 +1,106 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Examples
+
+## Create table
+### From file
+```ruby
+table = Arrow::Table.load('data.arrow')
+table = Arrow::Table.load('data.csv', format: :csv)
+table = Arrow::Table.load('data.parquet', format: :parquet) # requires 'red-parquet' gem
+```
+### From Ruby hash
+Types will be detected automatically
+```ruby
+table = Arrow::Table.new('name' => ['Tom', 'Max'], 'age' => [22, 23])
+```
+### From String
+Suppose you have your data available via Http. Let's connect to demo Clickhouse db. See https://play.clickhouse.com/ for details
+```ruby
+params = {
+  query: "SELECT WatchID as watch FROM hits_v1 LIMIT 10 FORMAT Arrow",
+  user: "playground",
+  password: "clickhouse",
+  database: "datasets"
+}
+uri = URI('https://play-api.clickhouse.com:8443')
+uri.query = URI.encode_www_form(params)
+resp = Net::HTTP.get(uri)
+table = Arrow::Table.load(Arrow::Buffer.try_convert(resp))
```

Comment:
```suggestion
table = Arrow::Table.load(Arrow::Buffer.new(resp))
```

BTW, we should add support for `Arrow::Table.load(URI("https://..."))` style. Could you open a Jira issue for it?

Reply:
I like the idea of supporting urls, but does `URI` gives us enough flexibility? What if we need to do POST request? And send some Authorization headers? I'm not sure if Arrow ecosystem has something builtin. Maybe we can use subset of `Net::HTTPRequest`, like `Net::HTTP::Get`?

Reply:
Apache Arrow C++ provides filesystem module: https://github.com/apache/arrow/tree/master/cpp/src/arrow/filesystem
It supports some HTTP based data source such as Amazon S3 and Google Cloud Storage.

I think that `URI("https://...")` is convenient for simple (and many?) use cases. How about thinking new API when we have use cases that require complex HTTP use such as POST and Authorization headers? It may be better that these complex use cases are implemented as a new filesystem.

Reply:
Makes sense for me.
https://issues.apache.org/jira/browse/ARROW-14562 Ticket to support `URI`
https://issues.apache.org/jira/browse/ARROW-14563 Ticket to support `S3FileSystem`, great to have feature